### DISCUSSION TITLE
Class instead struct for reducer

### DISCUSSION THREAD

**Discussion by ArmenNik** [pointfreeco/swift-composable-architecture#2406] [Q&A] [answered]
Hello guys, does anyone have tried to user class instead of struct to conform ReducerProtocol
Using a class for your reducer allow me have inheritance and curate some base for app
Is it good idea?

Thanks in advance

**Comment by mbrandonw** [comment, marked as answer]
Hi @ArmenNik, generally you do not want your reducer to be a class. It is very likely to cause some extremely strange behavior that will be difficult to track down. There are other ways of sharing logic, such as using reducer methods.

**Reply by jshier** [reply]
> Hi @ArmenNik, generally you do not want your reducer to be a struct.

May want to update the accepted answer to say "class" here.

**Reply by mbrandonw** [reply]
Oops, thanks @jshier, I meant _class_:

> Hi @ArmenNik, generally you do not want your reducer to be a class.

**Comment by ArmenNik** [comment]
Hi @mbrandonw  thanks for answer
